packages/flutter/lib/src/painting/colors.dart

```diff
@@ -193,7 +227,177 @@ class HSVColor {
   int get hashCode => hashValues(alpha, hue, saturation, value);
 
   @override
-  String toString() => 'HSVColor($alpha, $hue, $saturation, $value)';
+  String toString() => '$runtimeType($alpha, $hue, $saturation, $value)';
+}
+
+/// A color represented using [alpha], [hue], [saturation], and [lightness].
+///
+/// An [HSLColor] is represented in a parameter space that's motivated by human
+/// perception. The representation is useful for some color computations (e.g.,
+/// rotating the hue through the colors of the rainbow).
+///
+/// See also:
+///
+///   * [HSVColor], a color where the brightness is represented by value
+///     instead of lightness.
+@immutable
+class HSLColor {
+  /// Creates a color.
+  ///
+  /// All the arguments must not be null and be in their respective ranges. See
+  /// the fields for each parameter for a description of their ranges.
+  const HSLColor.fromAHSL(this.alpha, this.hue, this.saturation, this.lightness)
+    : assert(alpha != null),
+      assert(hue != null),
+      assert(saturation != null),
+      assert(lightness != null),
+      assert(alpha >= 0.0),
+      assert(alpha <= 1.0),
+      assert(hue >= 0.0),
+      assert(hue <= 360.0),
+      assert(saturation >= 0.0),
+      assert(saturation <= 1.0),
+      assert(lightness >= 0.0),
+      assert(lightness <= 1.0);
+
+  /// Creates an [HSLColor] from an RGB [Color].
+  ///
+  /// This constructor does not necessarily round-trip with [toColor] because
+  /// of floating point imprecision.
+  factory HSLColor.fromColor(Color color) {
+    final double red = color.red / 0xFF;
+    final double green = color.green / 0xFF;
+    final double blue = color.blue / 0xFF;
+
+    final double max = math.max(red, math.max(green, blue));
+    final double min = math.min(red, math.min(green, blue));
+    final double delta = max - min;
+
+    final double alpha = color.alpha / 0xFF;
+    final double hue = _getHue(red, green, blue, max, delta);
+    final double lightness = (max + min) / 2.0;
+    // Saturation can exceed 1.0 with rounding errors, so clamp it.
+    final double saturation = lightness == 1.0
+      ? 0.0
+      : (delta / (1.0 - (2.0 * lightness - 1.0).abs())).clamp(0.0, 1.0);
+    return new HSLColor.fromAHSL(alpha, hue, saturation, lightness);
+  }
+
+  /// Alpha, from 0.0 to 1.0. The describes the transparency of the color.
+  /// A value of 0.0 is fully transparent, and 1.0 is fully opaque.
+  final double alpha;
+
+  /// Hue, from 0.0 to 360.0. Describes which color of the spectrum is
+  /// represented. A value of 0.0 represents red, as does 360.0. Values in
+  /// between go through all the hues representable in RGB.
+  final double hue;
+
+  /// Saturation, from 0.0 to 1.0. This describes how colorful the color is.
+  /// 0.0 implies a shade of grey, and 1.0 implies a color as vibrant as that
+  /// hue gets.
+  final double saturation;
```

Comment:
this is exactly the same definition as for HSV, but the math is different, which doesn't make sense to me.